client/hwlib/src/collectors/os_info.rs

```diff
@@ -91,11 +91,26 @@ impl KernelPackage {
     }
 }
 
-pub(crate) fn get_architecture(runner: &impl CommandRunner) -> Result<String> {
-    let arch = runner
-        .run_command(DPKG, &["--print-architecture"])
-        .with_context(|| "cannot determine system architecture")?;
-    Ok(arch.trim().to_owned())
+pub(crate) fn to_debian_architecture(arch: &str) -> Result<&str> {
+    match arch.trim() {
+        "aarch64" => Ok("arm64"),
```

Comment:
The many `Ok` introduce a lot of noise here, I think it would be cleaner to re-introduce the `let` binding and explicitly return in the error case, i.e.
```rust
let ret = match arch.trim() {
    "arm" => "armhf",
    ...
    _ => return Err(...), 
};
Ok(ret)
```